apps/api/src/auth/user_identity.py

```diff
@@ -0,0 +1,157 @@
+import os
+from datetime import datetime, timedelta, timezone
+from typing import Annotated, Any, Union
+
+from fastapi import Cookie, HTTPException, Response, status
+from fastapi.testclient import TestClient
+from jose import JWTError, jwt
+from pydantic import BaseModel, EmailStr
+
+JWT_ALGORITHM = "HS256"
+JWT_SECRET = os.getenv("JWT_SECRET", "")
+
+COOKIE_NAME = "irvinehacks_auth"
+
+
+class User(BaseModel):
+    uid: str
+    email: EmailStr
+
+    def __str__(self) -> str:
+        return self.uid
+
+
+class NativeUser(User):
+    ucinetid: str
+    display_name: str
+    affiliations: list[str]
+
+    def __init__(
+        self, *, ucinetid: str, display_name: str, email: str, affiliations: list[str]
+    ):
+        uid = f"edu.uci.{ucinetid}"
+        super().__init__(  # type: ignore[call-arg]
+            uid=uid,
+            ucinetid=ucinetid,
+            display_name=display_name,
+            email=email,
+            affiliations=affiliations,
+        )
```

Comment:
Question: Is the reason why you ignored here because the additional arguments are needed to pass into the `BaseModel` constructor (in order to generate this model properly)?

Reply:
mypy didn't complain about this last year, but as of mypy v1.1.1, possibly because of the added support for `dataclass_transform`, we'll get error messages about the extra keyword arguments
> error: Unexpected keyword argument `ucinetid` for `__init__` of `User`  [[call-arg](https://mypy.readthedocs.io/en/latest/_refs.html#code-call-arg)]

Pydantic will raise other errors if I omit the extra attributes and try to assign them separately. There probably is a better way to do this, but I couldn't find such.

Reply:
Although not perfect, it seems using the [mypy plugin for pydantic](https://docs.pydantic.dev/latest/integrations/mypy/) _without_ [`init_forbid_extra`](https://docs.pydantic.dev/latest/integrations/mypy/#generate-a-signature-for-model__init__) will add kwargs as `Any` to `super().__init__`
> Revealed type is "def (*, uid: builtins.str, email: builtins.str, **kwargs: Any)"

Reply:
Awesome, thanks for finding a workaround for this.